packages/calypso-e2e/src/lib/pages/cart-checkout-page.ts

```diff
@@ -42,4 +73,105 @@ export class CartCheckoutPage {
 		await this.page.click( selectors.removeCartItemButton( cartItemName ) );
 		await this.page.click( selectors.modalContinueButton );
 	}
+
+	/**
+	 * Obtains the text content of the payment button.
+	 *
+	 * @returns {string} Content of the payment button.
+	 */
+	async getPaymentButtonText(): Promise< string > {
+		const elementHandle = await this.page.waitForSelector( selectors.paymentButton );
+		return await elementHandle.innerText();
+	}
+
+	/**
+	 * Apply a coupon.
+	 *
+	 * If a coupon code is supplied as parameter, this method will apply the supplied code.
+	 * Otherwise, this method will apply the default coupon code found in the configuration.
+	 *
+	 * @param {string} coupon Custom coupon code.
+	 */
+	async enterCouponCode( coupon?: string ): Promise< void > {
+		await this.page.click( selectors.couponCodeInputButton );
+
+		const couponCode =
+			coupon !== undefined ? coupon : ( DataHelper.config.get( 'testCouponCode' ) as string );
+
+		await this.page.fill( selectors.couponCodeInput, couponCode );
+		await this.page.click( selectors.couponCodeApplyButton );
+		await this.page.click( selectors.dismissBanner );
+	}
+
+	/**
+	 * Returns the total amount of items in cart.
+	 *
+	 * @returns {string} Total value of items in cart.
+	 */
+	async getCheckoutTotalAmount(): Promise< number > {
+		const elementHandle = await this.page.waitForSelector(
+			selectors.totalAmount( getTargetDeviceName() )
+		);
+		const stringAmount = await elementHandle.innerText();
+		const parsedAmount = stringAmount.replace( /,/g, '' ).match( /\d+\.?\d*/g );
+		if ( ! parsedAmount?.length ) {
+			throw new Error( 'Unable to locate or parse cart amount.' );
+		}
+		return parseInt( parsedAmount.pop() as string );
+	}
+
+	/**
+	 * Enter billing/tax details.
+	 *
+	 * @param param0 Object implementing the PaymentDetails interface.
+	 * @param {string} param0.postalCode Postal code of the user.
+	 * @param {string} param0.countryCode Country of the purchaser in ISO 3166-1 Alpha 2 format.
+	 */
+	async enterBillingDetails( { postalCode, countryCode }: PaymentDetails ): Promise< void > {
+		await this.page.fill( selectors.postalCode, postalCode );
+		await this.page.selectOption( selectors.countryCode, countryCode );
+		await this.page.click( selectors.submitBillingInformationButton );
+	}
+
+	/**
+	 * Enter payment details.
+	 *
+	 * @param param0 Object implementing the PaymentDetails interface.
+	 * @param {string} param0.cardHolder Credit card holder name.
+	 * @param {string} param0.cardNumber Credit card number.
+	 * @param {string} param0.expiryMonth Credit card expiry month.
+	 * @param {string} param0.expiryYear Credit card expiry year.
+	 * @param {string} param0.cvv Credit card CVV value.
+	 */
+	async enterPaymentDetails( {
+		cardHolder,
+		cardNumber,
+		expiryMonth,
+		expiryYear,
+		cvv,
+	}: PaymentDetails ): Promise< void > {
```

Comment:
Nit incoming! Since you have an interface for this, I think it would be more readable to just take a single named parameter of the `PaymentDetails` type, and then you can destructure it in the function. That should simplify the signature / header comments a good bit.

Reply:
Fixed in e16661468d2dcd16127a4850c2cf7e0fc289e78c.